### ISSUE TITLE
04_advanced_rag_techniques : how to use json file in the documents attribuition

### ISSUE THREAD

**Issue by amscosta** [benman1/generative_ai_with_langchain#96]
### Steps to reproduce

Hello guys:
From the above listed notebook , I want to adapt the cell:
documents = [
    Document(
        page_content="The transformer architecture was introduced in the paper 'Attention is All You Need' by Vaswani et al. in 2017.",
        metadata={"source": "Neural Network Review 2021", "page": 42}
    ),
    Document(
        page_content="BERT uses bidirectional training of the Transformer, masked language modeling, and next sentence prediction tasks.",
        metadata={"source": "Introduction to NLP", "page": 137}
    ),
    Document(
        page_content="GPT models are autoregressive transformers that predict the next token based on previous tokens.",
        metadata={"source": "Large Language Models Survey", "page": 89}
    )
]

For using a json file like below:
[
  {
    "content": "Psychrometrics uses thermodynamic properties to analyze conditions and processes involving moist air. This chapter discusses perfect gas relations and their use in common heating, cooling, and humidity control problems.",
    "metadata": {
      "source_pdf": "chapter1.pdf",
      "page": 42,
      "topic": "psychrometrics"
    }
  },
  {
    "content": "Heat transfer is energy transferred because temperature difference. Energy moves from a higher-temperature region to a lower-temperature region by one or more of three modes: conduction, radiation, and convection. This chapter presents elementary principles of single-phase heat transfer, with emphasis on HVAC applications. Boiling and condensation are discussed in Chapter 5.",
    "metadata": {
      "source_pdf": "chapter4.pdf",
      "page": 137,
      "topic": "heat_transfer"
    }
  },
]

Hopefully I am being enoughly clear.
Any helpful thoughts are very welcome.

### Branch

None

**Comment by benman1** [comment]
Hi @amscosta 
You can do something like this to convert JSON entries to LangChain Document objects:
```python
documents = [
    Document(
        page_content=item["content"],
        metadata=item.get("metadata", {})
    )
    for item in json_docs
]
```

Does this answer your question?

**Comment by benman1** [comment]
Closing this
